bravenewcoin-vwap/adapter.js

```diff
@@ -0,0 +1,96 @@
+const { Requester, Validator } = require('@chainlink/external-adapter')
+
+const customParams = {
+  symbol: ['base', 'from', 'coin', 'symbol', 'assetId', 'indexId', 'asset'],
+  indexType: false,
+  timestamp: false
+}
+
+const host = 'bravenewcoin.p.rapidapi.com'
+
+const authenticate = () => {
+  return new Promise((resolve, reject) => {
+    Requester.request({
+      method: 'POST',
+      url: `https://${host}/oauth/token`,
+      headers: {
+        'content-type': 'application/json',
+        'x-rapidapi-host': host,
+        'x-rapidapi-key': process.env.API_KEY,
+        accept: 'application/json',
+        useQueryString: true
+      },
+      data: {
+        audience: 'https://api.bravenewcoin.com',
+        client_id: process.env.CLIENT_ID,
+        grant_type: 'client_credentials'
+      }
+    }).then(response => resolve(response.data.access_token))
+      .catch(error => reject(error))
+  })
+}
```

Comment:
Using Promises directly becomes complicated fairly quickly, that's why `async/await` was introduced and IMHO it makes code a lot easier to reason about.

We could rewrite this snippet using `async/await`:
```suggestion
const authenticate = async () => {
  const response = await Requester.request({
    method: 'POST',
    url: `https://${host}/oauth/token`,
    headers: {
      'content-type': 'application/json',
      'x-rapidapi-host': host,
      'x-rapidapi-key': process.env.API_KEY,
      accept: 'application/json',
      useQueryString: true
    },
    data: {
      audience: 'https://api.bravenewcoin.com',
      client_id: process.env.CLIENT_ID,
      grant_type: 'client_credentials'
    }
  })
  return response.data.access_token
}
```